tsconfig.build.json

```diff
@@ -5,15 +5,16 @@
     "noImplicitAny": true,
     "strictNullChecks": true,
     "noFallthroughCasesInSwitch": true,
-    "module": "commonjs",
-    "target": "es5",
-    "lib": ["es2015"],
+    "lib": ["ES2022"],
+    "module": "node16",
+    "target": "ES2022",
```

Comment:
Just to double-check: 
- `"module": "node16"` means that we still transpile to CJS, correct? If we were to transpile to ESM, this would be breaking and we also need to adjust the package.json entry points accordingly.
- `"target": "es2022"`: is es2022 supported by Node 18?

Reply:
Took these from https://github.com/microsoft/TypeScript/wiki/Node-Target-Mapping#node-18

Don't think it affects CJS/ESM as I need working CJS bundles for #758.